edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ServiceIdentityTree.cs

```diff
@@ -95,11 +95,16 @@ public async Task<Option<string>> GetAuthChain(string id)
         {
             Preconditions.CheckNonWhiteSpace(id, nameof(id));
 
-            // Auth-chain for the root (when there aren't any nodes in the tree yet)
+            // Auth-chain for the acting device (when there aren't any nodes in the tree yet)
             if (id == this.actorDeviceId)
             {
                 return Option.Some(this.actorDeviceId);
             }
+            else if (id == this.actorDeviceId + "/$edgeHub")
```

Comment:
Why do you need to treat EdgeHub on the device differently? (Why can't it be handled by the loop below?) Is it because we need this for EdgeHub even if it hasn't been populated yet?

Nit: We should use a const for $edgeHub

Reply:
yes, it's because we make the initial upstream call to get the scope as $edgeHub, and the tree isn't populated yet at that point.